components/StdHeader.vue

```diff
@@ -0,0 +1,99 @@
+              </div>
+              <button type="submit" class="p-2.5 ml-3 text-sm font-medium text-white bg-blue-700 rounded-lg border border-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">
+                <svg-icon type="mdi" :path="path" class="h-7 w-7"></svg-icon>
+              </button>
+             </form>
+
+             <span class="mt-7 mx-12 underline">Stats</span>
+             
+            <label class="relative inline-flex items-center cursor-pointer mt-2">
+              <input type="checkbox" value="" class="sr-only peer">
+              <div class="w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:left-[2px] after:top-[25px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600"></div>
+              <span class="ml-3 font-medium text-gray-500 dark:text-gray-300">Classify Words</span>
+            </label>
+
+            <span class="flex mt-7 mx-10 text-gray-800">mkapoor@idir</span>
+
+            <div class="divider"></div>
+
+            <div class="mx-10 mt-7 text-blue-800">
+              <a href="#" @click="logout">Log Out</a>
+             </div>
+          </div>
+
+          <div class="flex ml-auto mx-10 mt-7 text-blue-800" v-if="!authModule.isAuthenticated">
+              <a href="#" @click="login">Login</a> 
+             </div>
+
+      </div>
+  </nav>
+</template>
+
+<script setup>
+import SvgIcon from '@jamescoyle/vue-icon';
+import { mdiMagnify } from '@mdi/js';
+import { useAuthStore } from '~/store/auth';
+import KeycloakService from '~/public/keycloak/keycloak'
+
+const authModule = useAuthStore()
+const path = mdiMagnify
+
+const login = async () => {
+  // If the user is already authenticated, do nothing
+  if (authModule.isAuthenticated) return
+
+  try {
+    // Start the token initialization process and wait for it to finish
+    await KeycloakService.initializeToken(authModule, true, !authModule.isAuthenticated);
+
+    // Token should now be generated
+    await KeycloakService.initSession()
+
+    
+  } catch (err) {
+    if (err?.message !== 'NOT_AUTHENTICATED') {
+      console.error(err)
+      // Handle the error appropriately, possibly by showing an error message to the user
+    }
+  }
+}
+
+
+const logout = async () => {
+  // If the user is already logged out, do nothing
+  if (!authModule.isAuthenticated) return
+
+  try {
+    await KeycloakService.logout("http://localhost:8080/")
+  } catch (err) {
+    if (err?.message !== 'LOGOUT FAILED') {
+      console.error(err)
+      // Handle the error appropriately, possibly by showing an error message to the user
```

Comment:
How are you keeping track of things that still need to be implemented?

Reply:
We're making tickets on the SRE board to keep a track. Once this PR is merged, ills make a new one for for the tests and then one for search name requests UI.

Reply:
OK re: this item to do.

As for separate tickets for the unit tests, what I've seen is that those tickets are always de-prioritized in favour of new feature work, and then you end up with no tests. Therefore, I think unit tests should be part of the feature work.

Reply:
Understood, I'll be including the unit tests within this same PR. Regarding the 'to do' item you mentioned, could you please provide some clarification on what exactly you meant by that?

Reply:
I was saying OK to "We're making tickets on the SRE board to keep a track. ".

All good, thanks. Resolving.

Reply:
Or maybe add unit tests in a new PR against the same ticket? Just as long as we don't bury the unit test work :)